### ISSUE TITLE
47.6 Omnimechs not showing refit packages and MekHQ freezes when custom refits used

### ISSUE THREAD

**Issue by KiloWhiskey1** [MegaMek/mekhq#1748]
Attempting to change out a Black Hawk A to a Prime. When clicking the refit button in the hanger no refit options are shown. When attempting a custom refit MekHQ will freeze upon clicking begin refit. Refit then appears in the custom mech files, but still not available to use as a refit in MekHQ.
I have tried changing acquisition settings to acquiring clan parts with no change in issue.

### Environment
47.6
Windows
 1.8.0_231

### Description
Attempting to change out a Black Hawk A to a Prime. When selecting the refit option in the hanger no refit options are shown. When attempting a custom refit MekHQ will freeze upon clicking begin refit. Refit then appears in the custom mech files, but still not available to use as a refit in MekHQ.
I have tried changing acquisition settings to acquiring clan parts with no change in issue.

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/4599317/mekhqlog.txt)

[customs.zip](https://github.com/MegaMek/mekhq/files/4599309/customs.zip)

[RedRum30530203.zip](https://github.com/MegaMek/mekhq/files/4599305/RedRum30530203.zip)

**Comment by sixlettervariables** [comment]
No issues refitting a stock Black Hawk (Nova) A to a Prime using the latest dev snapshot.

**Comment by sixlettervariables** [comment]
@Windchild292 unrelated:
```
16:49:00,489 WARN [mekhq.campaign.io.CampaignXmlParser] {SwingWorker-pool-1-thread-1} 
migrateAncestorNodes : Person with id null does not exist, skipping adding Genealogy for them.

16:49:00,489 WARN [mekhq.campaign.io.CampaignXmlParser] {SwingWorker-pool-1-thread-1} 
migrateAncestorNodes : Person with id a1a5c398-482e-4c5d-ae07-5d537da67b0f does not exist, skipping adding Geneology for them.

16:49:00,489 WARN [mekhq.campaign.io.CampaignXmlParser] {SwingWorker-pool-1-thread-1} 
migrateAncestorNodes : Person with id null does not exist, skipping adding Genealogy for them.
```
Wondering if this is something we need fixed, or if this is expected for stable?

**Comment by Windchild292** [comment]
@sixlettervariables 
That is working as intended, trimming out erroneous ancestry data that was never properly cleaned out.

**Comment by sixlettervariables** [comment]
@KiloWhiskey1 because that mech is a Clan Omni, based on your Campaign option disallowing mixed tech, you cannot choose a refit kit for it. That part is working as designed.

**Comment by sixlettervariables** [comment]
@KiloWhiskey1 okay, as far as doing the refit in the lab...well it doesn't "hang" entirely. It just takes >500 seconds on my machine. Here are the highlights as to why it is so slow:
1. The new unit gets saved and the MechSummary/Units cache gets rebuilt. This takes ~2.5 minutes on my machine.
    ```
    mekhq.campaign.parts.Refit.saveCustomization ()	158,287 ms (46.3%)	158,287 ms (46.3%)
    	megamek.common.MechSummaryCache.loadMechData ()	152,589 ms (44.6%)	152,589 ms (44.6%)
    		megamek.common.MechSummaryCache.loadMechsFromDirectory ()	58,008 ms (17%)	58,008 ms (17%)
    	megamek.common.MechSummaryCache.saveCache ()	35,397 ms (10.3%)	35,397 ms (10.3%)
    ```
2. Then it has to find the parts, and since you use Planetary Acquistion _and_ you can't buy clan parts, that takes another ~30 seconds:
    ```
    mekhq.campaign.Campaign.goShoppingByPlanet ()	23,799 ms (7%)	23,799 ms (7%)
    ```
3. Then it takes over a minute to start refitting the unit:
    ```
    mekhq.campaign.Campaign.refit ()	75,307 ms (22%)	75,307 ms (22%)
    	mekhq.campaign.parts.Refit.succeed ()	47,694 ms (13.9%)	47,694 ms (13.9%)
    		mekhq.campaign.parts.Refit.complete ()	47,694 ms (13.9%)	47,694 ms (13.9%)
    			mekhq.campaign.Campaign.reloadGameEntities ()	28,905 ms (8.4%)	28,905 ms (8.4%)
    			mekhq.MekHQ.triggerEvent ()	18,287 ms (5.3%)	18,287 ms (5.3%)
    				mekhq.gui.WarehouseTab.filterParts ()	18,287 ms (5.3%)	18,287 ms (5.3%)
    ```
4. Finally, it takes a long time to _graphically_ refresh every tab in MHQ which I could not capture in my perf traces.

Long story short, the behavior you're seeing is "by design" but I am going to keep this open as a performance related issue regarding custom refits.
